src/nvim/ex_getln.c

```diff
@@ -1834,15 +1839,26 @@ getexmodeline (
         } else {
           p[line_ga.ga_len] = NUL;
           indent = get_indent_str(p, 8, FALSE);
+          if (indent == 0) {
+            continue;
+          }
           --indent;
           indent -= indent % get_sw_value(curbuf);
         }
-        while (get_indent_str(p, 8, FALSE) > indent) {
-          char_u *s = skipwhite(p);
 
-          memmove(s - 1, s, line_ga.ga_len - (s - p) + 1);
-          --line_ga.ga_len;
+        // Delete trailing leading whitespaces until enough indentation
+        // is removed.
```

Comment:
Is this clearer?
`Remove a big enough suffix of the whitespace prefix for the Ex command line to reduce the line indentation.`